### PR TITLE
Embed Miscellaneous Improvements and Fixes

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

__Please don't focus on number of files changes as there are many snapshot files 😃__
Complete Demo Video - https://www.loom.com/share/2dc8c3d55517420a8021bfd137bb0152
A Video showing Direct Link Bookings(No change expected in it)- https://www.loom.com/share/76a35946834343619f93ad615318e5a2

Demos - Payment Page in Embed 

**Fixes**
- Right now border has been changed to `rounded-md` instead of `rounded-sm` to make the popup look cleaner. It seems okay to me to have the same in the direct link as well. [Needs Review by Team][Ensure that everywhere it's rounded-md and not rounded-lg]
- Add missing custom branding support in Event Types Listing. [TODO: Verify the impact on the direct link as well]
- Fixes the issue where if the `email` contains `embed`, app wrongly thinks that it is in embed mode.
- Payments Page UI Fixes and Stripe Iframe positioning hack to show it properly in embed iframe.
- Fixes a UI issue in embed when there are too many event-types
- Fixed a very random bug when the dark theme was being applied instead of the light theme.

**Improvements**
- Embed is fully responsive now with the new width handling approach[An attempt at documenting the approach [here](https://www.notion.so/calcom/Cal-Embed-Concepts-c27e122d07084806829a6755b8eea934) ]
- Use `app.cal.com` till the page reload problem is solved. It improves performance
- Automatically scroll iframe to top to handle drastic changes in heights of the embed iframe.
- Tailwind Support in Web Components - Restructured the files for this, HTML being stored in a separate file now so that tailwindcss can run on it and extract optimized tailwind.generated.css which can be added just once in embed.js. Note: Regular tailwind integration won't work with Web Components because generated styles can't go in body, they would have to go in Shadow DOM.
- Close ModalBox/Popup with Escape Key
- NPM embed-react package can now be used in a non TypeScript project as well. It now publishes .js files instead of .ts files along with declaration files.
- Floating Embed UI Improvements

## Tests
- Tests Setup for embed-react using the config from embed-core.
- A basic snapshot-based test for` embed-react`
- More tests for embed-core, to verify the booking data in DB. Now it tests if prefilled guests are correctly stored in DB. This was a bug earlier in embed where an empty guest was being stored in DB.
**NOTE: Embed Tests still don't run in CI, but they can be run locally to ensure nothing major breaks in Safari,Firefox and Chrome**

## Pending Bugs Verification
- Cal React component if used with config theme "light" causing duplicate `cal-inline` addition.
- Loader position in inline modes with center and left alignment

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test on an existing customer's website.

## Checklist

- I haven't performed a self-review of my own code and corrected any misspellings

### CODE REVIEW

**Comment by zomars** [comment]
@hariombalhara regarding those snapshot files. They will only work on local tests since GitHub actions don't use `darwin` but linux. Two possible solutions:

- Make the snapshots without the os suffix
- Generate the snapshots using linux to CI can make use of them

**Comment by hariombalhara** [comment]
Ignoring OS might fail tests as snaphots actually vary. 
May be we can introduce a job(manually triggered) that can be run whenever we want to introduce/update snapshots. It can commit those as well.

**Review by PeerRich — APPROVED** [review]

**Comment by hariombalhara** [comment]
Playwright has a flag to generate snapshots only(--update-snapshots), it doesn't look for actual snapshots and just generates them. 
Once they are generated, it's just about an automated git add, commit and push

File: apps/docs/pages/integrations/embed.mdx [hunk 1]
```diff
@@ -72,18 +68,20 @@ Show the embed inline inside a container element. It would take the width and he
...
+<script>
+  Cal("inline", {
+    elementOrSelector: "Your Embed Container Selector Path", // You can also provide an element directly
+    calLink: "jane", // The link that you want to embed. It would open https://cal.com/jane in embed
+    config: {
+      name: "John Doe", // Prefill Name
+      email: "johndoe@gmail.com", // Prefill Email
+      notes: "Test Meeting", // Prefill Notes
+      guests: ["janedoe@gmail.com", "test@gmail.com"], // Prefill Guests
+      theme: "dark", // "dark" or "light" theme
```
Comment (alannnc) [inline apps/docs/pages/integrations/embed.mdx:81]: I would avoid valid emails @gmail.com here `@cal.com` maybe?

File: packages/embeds/embed-core/index.html [hunk 2]
```diff
@@ -5,6 +5,14 @@
       if (!location.search.includes("nonResponsive")) {
         document.write('<meta name="viewport" content="width=device-width"/>');
       }
+      (()=> {
+        const u = new URL(document.URL);
```
Comment (alannnc) [inline packages/embeds/embed-core/index.html:9]: I would suggest more elaborated variable names, e.g. `const url`

File: packages/embeds/embed-core/index.html [hunk 3]
```diff
@@ -5,6 +5,14 @@
...
+        const only = u.searchParams.get("only");
+        const ns = only ? only.replace("ns:",""): null
```
Comment (alannnc) [inline packages/embeds/embed-core/index.html:11]: what do `only` and `ns` do?

File: packages/embeds/embed-core/playwright/config/playwright.config.ts [hunk 4]
```diff
@@ -3,10 +3,10 @@ import * as path from "path";
 
 const outputDir = path.join("../results");
 const testDir = path.join("../tests");
-
+const QuickMode = process.env.QUICK === "true";
```
Comment (alannnc) [inline packages/embeds/embed-core/playwright/config/playwright.config.ts:6]: should we do camelCase here `quickMode` to match everything on the file?

**Review by alannnc — APPROVED** [review]
Code looks good, haven't tested it yet. Just some nit changes

**Comment by hariombalhara** [comment]
Thanks @alannnc, all of your comments were valid. Addresse them

**Pull request merged**